Documentation/contributing/development/introducing_new_crds.rst

```diff
@@ -0,0 +1,400 @@
+   +
+   +       return createUpdateCRD(
+   +               clientset,
+   +               BGPPoolCRDName,
+   +               constructV1CRD(k8sconstv2alpha1.BGPPName, ciliumCRD),
+   +               newDefaultPoller(),
+   +       )
+   +}
+   +
+    // createUpdateCRD ensures the CRD object is installed into the K8s cluster. It
+    // will create or update the CRD and its validation schema as necessary. This
+    // function only accepts v1 CRD objects, and defers to its v1beta1 variant if
+
+Getting your CRDs installed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your new CRDs must be installed into Kubernetes. This is controlled in
+the ``pkg/k8s/synced/crd.go`` file.
+
+Here is an example diff which installs the CRDs ``v2alpha1.BGPPName``
+and ``v2alpha.BGPPoolName``
+
+.. code:: diff
+
+   diff --git a/pkg/k8s/synced/crd.go b/pkg/k8s/synced/crd.go
+   index 52d975c449..10c554cf8a 100644
+   --- a/pkg/k8s/synced/crd.go
+   +++ b/pkg/k8s/synced/crd.go
+   @@ -42,6 +42,11 @@ func agentCRDResourceNames() []string {
+                   CRDResourceName(v2.CCNPName),
+                   CRDResourceName(v2.CNName),
+                   CRDResourceName(v2.CIDName),
+   +               CRDResourceName(v2.CIDName),
+   +               // TODO(louis) make this a conditional install
+   +               // based on --enable-bgp-control-plane flag
+   +               CRDResourceName(v2alpha1.BGPPName),
+   +               CRDResourceName(v2alpha1.BGPPoolName),
+           }
+
+Updating RBAC roles
+~~~~~~~~~~~~~~~~~~~
+
+Cilium is installed with a service account and this service account
+should be given RBAC permissions to access your new CRDs. The following
+files should be updated to include permissions to CRUD your new CRD.
+
+.. code:: text
+
+   install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml
+   install/kubernetes/cilium/templates/cilium-preflight/clusterrole.yaml
+
+Here is a diff of updating the Agent’s cluster role template to include
+our new BGP CRDs
+
+.. code:: diff
+
+   diff --git a/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml b/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   index 9878401a81..5ba6c30cd7 100644
+   --- a/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   +++ b/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   @@ -102,6 +102,8 @@ rules:
+      - ciliumlocalredirectpolicies/finalizers
+      - ciliumegressnatpolicies
+      - ciliumendpointslices
+   +  - ciliumbgppeeringpolicies
+   +  - ciliumbgploadbalancerippools
+      verbs:
+      - '*'
+    {{- end }}
+
+It’s important to note, neither the Agent nor the Operator installs
+these manifests to the Kubernetes clusters. This means when testing your
+CRD out the updated ``clusterrole`` must be written to the cluster
+manually.
+
+Andre provided a convenient script for this:
```

Comment:
```suggestion
André provided a convenient script for this:
```
Although I'm not sure we usually name people in the docs?

Reply:
Agreed, I'll remove the name from the documentation. This is not the right